Cabal/src/Distribution/Simple/InstallDirs.hs

```diff
@@ -47,6 +47,9 @@ module Distribution.Simple.InstallDirs (
         packageTemplateEnv,
         abiTemplateEnv,
         installDirsTemplateEnv,
+
+        maybeGetCabalDir,
```

Comment:
it's kind of sad that `Cabal` has this functionality.

I'd try to structure this so, that `cabal-install` handles all of backward-compat/migration, and `Cabal` the library can have defaults independent of the legacy.

Reply:
This is difficult because some of the stuff in InstallDirs influences the default configuration produced in cabal-install - essentially a kind of dynamic cyclic dependency.  I agree it would be better design, but InstallDirs is used many places in Cabal, and I doubt I have the knowledge (or frankly, inclination) to exfiltrate it.

Reply:
This is gone now.